scene/animation/animation_node_state_machine.cpp

```diff
@@ -479,10 +479,8 @@ float AnimationNodeStateMachinePlayback::process(AnimationNodeStateMachine *sm,
 		}
 	}
 
-	//compute time left for transitions by using the end node
 	if (sm->end_node != StringName() && sm->end_node != current) {
-
```

Comment:
This breaks stuff, time for transition needs to be computed for it to work properly.

Reply:
It may be bad for visualization, but it needs to work this way.

Reply:
> This breaks stuff, time for transition needs to be computed for it to work properly.

Are you sure? I didn't find any issue, can you tell me where it can break so I can take more care?

This change is not necessary for this pr then I reverted this part, but I've a question, I'm working on a small refactoring on state machine to allow transitions between nodes in different state machines (like unreal/unity), sub state machine will have by default an start/end node and these nodes will be a simple `AnimationRootNode`, for that case, I need to set `rem = 1` otherwise the sub state machine stops immediately. For this scenario I can make that change? Do you know another way?

See this example:
![peek 07-01-2019 14-51](https://user-images.githubusercontent.com/1387165/50781155-cb2d2b80-128b-11e9-8437-e6c2ec35e012.gif)

And when I change to `rem = 1` works fine:
![peek 07-01-2019 14-43](https://user-images.githubusercontent.com/1387165/50780855-23aff900-128b-11e9-8a03-92bafa293911.gif)

Reply:
I think I found a solution for this.